cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -63,25 +64,32 @@ public void classify(CloseableIterator<LogEvent> logIter, HttpResponder responde
         if (ProgramFailureException.class.getName().equals(throwableProxy.getClassName())) {
           errorMessage = throwableProxy.getMessage();
         }
+        if (Logging.VALIDATION_EXCEPTION_CLASS.equals(throwableProxy.getClassName())) {
```

Comment:
I am wondering if we should just do `FailureDetailsProvider.class.isAssignableFrom(Class.forName(throwableProxy.getClassName()))` with caching and exception handling (class not found is OK). Current way just fixates on 2 classes, everything else is generic

Reply:
ack, done